tests/unittests/test_datasource/test_azure.py

```diff
@@ -632,15 +651,106 @@ def test_get_resource_disk(self):
         dev = ds.get_resource_disk_on_freebsd(1)
         self.assertEqual("da1", dev)
 
-    @mock.patch(MOCKPATH + '_is_platform_viable')
-    def test_call_is_platform_viable_seed(self, m_is_platform_viable):
+    def test_not_is_platform_viable_seed_should_return_no_datasource(self):
         """Check seed_dir using _is_platform_viable and return False."""
         # Return a non-matching asset tag value
-        m_is_platform_viable.return_value = False
-        dsrc = dsaz.DataSourceAzure(
-            {}, distro=mock.Mock(), paths=self.paths)
-        self.assertFalse(dsrc.get_data())
-        m_is_platform_viable.assert_called_with(dsrc.seed_dir)
+        data = {}
+        dsrc = self._get_ds(data)
+        self.m_is_platform_viable.return_value = False
+        with mock.patch.object(dsrc, 'crawl_metadata') as m_crawl_metadata, \
+                mock.patch.object(dsrc, '_report_failure') as m_report_failure:
+            ret = dsrc.get_data()
+            self.m_is_platform_viable.assert_called_with(dsrc.seed_dir)
```

Comment:
`self.m_is_platform_viable` should be created with `autospec=True` per HACKING.md (search for "autospec" in https://cloudinit.readthedocs.io/en/latest/topics/hacking.html#testing).

Reply:
Yeah thanks for pointing out `autospec`. Yelp actually wrote a nice anecdote about how they encountered lots of issues due to this 😄  https://engineeringblog.yelp.com/2015/02/assert_called_once-threat-or-menace.html